### DISCUSSION TITLE
Bug in OpenAPI definition after adding view

### DISCUSSION THREAD

**Discussion by kvnam** [PostgREST/postgrest#2601] [Q&A] [answered]
Hey,

In my Postgres DB I have a projects table, which has a foreign key `place_id` to the `places` table.

The API spec response, for the `projects` definition, correctly returned in this case.

```
"place_id": {
        "description": "Note:\nThis is a Foreign Key to `places.id`.<fk table='places' column='id'/>",
        "format": "bigint",
        "type": "integer"
 },
```
I then added a view

```
create view my_view as select pl.col1, pl.col2, pl.col3, pl.id, array_agg(f.path) as paths from files f
inner join projects_files pf on pf.file_id = f.id
left join projects p on p.id = pf.project_id
inner join places pl on pl.id = p.place_id
group by pl.col1, pl.id;
```

After adding this view the API spec now returns

```
"place_id": {
     "description": "Note:\nThis is a Foreign Key to `my_view.id`.<fk table='my_view' column='id'/>",
     "format": "bigint",
     "type": "integer"
 },
```

I rely on this response to build values in my app, and this is causing things to break. The foreign key on `projects` is still obviously pointing to `places` and not to this view. Can someone please let me know why this could be happening and what I can do to fix this? I need the id from `places` in my view, so that is unavoidable, is there a workaround? Has anyone else faced this issue?

Look forward to any help here!

**Comment by steve-chavez** [comment, marked as answer]
Hello. This is being addressed on https://github.com/PostgREST/postgrest/issues/2619

**Reply by kvnam** [reply]
Great, thanks @steve-chavez !
